### PR TITLE
fix: binaries

### PR DESCRIPTION
**Pull request by n1ru4l**
Closes https://github.com/dungeon-revealer/dungeon-revealer/issues/425

### CODE REVIEW

**Comment by n1ru4l** [comment]
@maxb2 I think the easiest way would be moving all the zipping stuff up to the job_build stage, as we also have access to the node_modules there. Then we could use the script (https://github.com/dungeon-revealer/dungeon-revealer/pull/427/commits/a7a6b90c4ce3062313febf04e276a7468c068ae7) for copying and bundling the `node_sqlit3.node` file over.

**Comment by n1ru4l** [comment]
I pushed a test tag https://github.com/dungeon-revealer/dungeon-revealer/runs/660552702?check_suite_focus=true

currently failing due to RUNNER_OS not being lowercased

**Comment by maxb2** [comment]
> @maxb2 I think the easiest way would be moving all the zipping stuff up to the job_build stage, as we also have access to the node_modules there. Then we could use the script ([a7a6b90](https://github.com/dungeon-revealer/dungeon-revealer/commit/a7a6b90c4ce3062313febf04e276a7468c068ae7)) for copying and bundling the `node_sqlit3.node` file over.

We'd have to make sure the zipping command works across Windows, OSX, and Linux. That's why I had it in the later job. I'll see what I can do.

**Comment by maxb2** [comment]
We should have a `DOCKER_REPO` secret set. That way we can set the docker repo in our forks to our personal repo. We could do all the actions testing in our own forks easily.

**Comment by n1ru4l** [comment]
The windows hassle is real 😖

**Comment by maxb2** [comment]
It doesn't look like WSL is [installed on their runners](https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md). 

> The windows hassle is real confounded

Try setting the shell to bash
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun

**Comment by n1ru4l** [comment]
yeah we need some workaround for zip x_x

**Comment by maxb2** [comment]
When compiling, we could output the binaries into subdirectories based on the OS. Then the artifact would have the structure:
```
bin/
  macos/
  windows/
  linux/
```
Then the next job will zip things in the subdirectories. We probably just have to make a separate job for Windows that uses PowerShell.

**Comment by n1ru4l** [comment]
Here we go again: https://github.com/dungeon-revealer/dungeon-revealer/runs/661306996?check_suite_focus=true

I tried some PowerShell: 9d6cc02

**Comment by n1ru4l** [comment]
The error log is not really helpful :D

**Comment by n1ru4l** [comment]
Okay the only thing failing rn is the windows build, but according to the docs the PowerShell is used by default and the used commands (alias) should exist. @maxb2 any ideas?

**Comment by maxb2** [comment]
I think it's failing at `mkdir`. I'll take a shot at it tonight.

**Comment by n1ru4l** [comment]
Was an issue with the mv command, it is currently building arm. Feel free to check whether the GitHub Release is created correctly afterwards :D I will go to sleep now and check again tomorrow morning.

**Comment by n1ru4l** [comment]
https://github.com/actions/download-artifact 

We must use v2 instead of v1 for downloading all artifacts at once

**Comment by maxb2** [comment]
I think we can combine this back into a matrix. A lot of the code is the same.

File: .github/workflows/build-binaries.yml [hunk 1]
```diff
@@ -40,33 +150,50 @@ jobs:
...
+      - name: Set environment
+        run: |
+          node ./scripts/set-git-tag-env.js
+
       - name: Build arm
         run: |
-            docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-            docker run --rm -v $(pwd):/usr/src/app/ arm32v7/node:12 /bin/sh -c 'cd /usr/src/app/; npm install; npm run compile:armv7'
+          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+          docker run --rm -v $(pwd):/usr/src/app/ arm32v7/node:12 /bin/sh -c 'cd /usr/src/app/; npm install; npm run build; node ./scripts/copy-node-bindings-path.js; npm run compile:armv7;'
```
Comment (maxb2) [inline .github/workflows/build-binaries.yml:165]: Is the build step required for compilation? It adds a considerable amount of time to the arm compilation.

[the remark this answers is not in the document]
Comment (n1ru4l) [reply .github/workflows/build-binaries.yml]: Tried to fix it, but it failed :D https://github.com/dungeon-revealer/dungeon-revealer/runs/662416413?check_suite_focus=true

[the remark this answers is not in the document]
Comment (maxb2) [reply .github/workflows/build-binaries.yml]: I use the environment variables `PUID` and `PGID` see [here](https://github.com/n1ru4l/dungeon-revealer/pull/2/commits/0ff4903909d09a4acc12cee27cdf35a0e82d1662). I've never been able to get `--user` to work the way I want 🤷

**Pull request merged**